### ISSUE TITLE
Short and long screenshots have different widths

### ISSUE THREAD

**Issue by bersbersbers** [rstudio/shinytest2#171]
```r
library(shiny)
library(shinytest2)

make_app <- function(n) {
  shinyApp(
    ui = HTML(paste(rep("X", n), collapse="<br>")),
    server = function(input, output) {}
  )
}

AppDriver$new(make_app(n = 1))$set_window_size(100, 100)$get_screenshot("snap1.png")
AppDriver$new(make_app(n = 9))$set_window_size(100, 100)$get_screenshot("snap9.png")
```

Both snapshots should be 100 pixels wide, but `snap9.png` comes out 85 pixels wide. 15 pixels seems to be the width of a vertical scrollbar. This is an issue in `chromote` (coming up shortly) which may require code changes in `shinytest2`.

Versions used: https://github.com/rstudio/chromote/commit/865e24e2270afab07e587909499c4e06f03bbedf, https://github.com/rstudio/shinytest2/commit/91ff4e58cad2d09e0e90f50e06911b1f47aa85d9
(in particular, it is independent of the fix to rstudio/chromote#81)

**Comment by wch** [comment]
I don't agree that both screenshots should necessarily be 100 pixels wide. The way it works is: there is a viewport size; the browser renders the page for that size viewport, and then it takes a screenshot of the page. What exactly happens in that last step is, of course, tricky. 

When a (normal) browser views a tall page, it adds a scrollbar and makes the page size narrower to accommodate. (Note that the behavior might be different on a Mac with auto-hiding scrollbars.) So the current screenshot behavior takes an image of the page that is the same size as what the browser says it is.

All that said, I don't think that the screenshot of the longer page should necessarily be narrower than 100 pixels either -- I'm just pointing out that the behavior is a choice, and that it's not that one is right and the other is wrong. I do think that a 100-pixel-wide screenshot is more convenient for many use cases.

One more note: If any changes are made here, they should be tested on:
* A Mac with auto-hiding scrollbars
* A Windows or Linux machine, or aMac with visible scrollbars

This is because auto-hiding scrollbars on a Mac work differently from the visible scrollbars -- the former get overlaid on the page and don't reduce the width.

**Comment by bersbersbers** [comment]
@wch I fully agree. I had understood the presence of `self$Emulation$setScrollbarsHidden(hidden = TRUE, wait_ = FALSE)` in
https://github.com/rstudio/chromote/blob/865e24e2270afab07e587909499c4e06f03bbedf/R/screenshot.R#L52 (which in other contexts *does* make the two viewports and snapshots 100 pixels wide, see https://github.com/rstudio/chromote/issues/84#issuecomment-1090653681) as an indication that this choice was to make the two snapshots the same width - or am I overlooking another reason for calling `setScrollbarsHidden`?

**Comment by bersbersbers** [comment]
I think the perfect solution would add some `hide_scrollbars_for_full_width = FALSE` argument to shinytest2`s `get_screenshot` which calls `chrome$Emulation$setScrollbarsHidden(hidden = TRUE)` before taking the snapshot. The ineffective call to `self$Emulation$setScrollbarsHidden` inside `chromote`'s `screenshot` could potentially be removed (but see https://github.com/rstudio/chromote/issues/84 for that).

**Comment by bersbersbers** [comment]
Here's another thought related to snapshot sizes. While the above thoughts are about having some mimum size, this example is about exceeding that minimum size. In this snippet, I had somewhat expected that `snap9.png` is made wide enough to accomodate all `X`, leading to snapshot more than 100 pixels wide:

```r
library(shiny)
library(shinytest2)

make_app <- function(n) {
  shinyApp(
    ui = HTML(paste(rep("XXX", n), collapse="")),
    server = function(input, output) {}
  )
}

AppDriver$new(make_app(n = 1))$set_window_size(100, 100)$get_screenshot("snap1.png")
AppDriver$new(make_app(n = 9))$set_window_size(100, 100)$get_screenshot("snap9.png")
```

(I had expected that due to the use of `captureBeyondViewport = TRUE` in https://github.com/rstudio/chromote/blob/1884e3bd5a4854832b44387a3cb3144c2ff757af/R/screenshot.R#L103)

I have played around with `selector`s a bit (HTML, body, ...) without much success so far.

**Comment by bersbersbers** [comment]
It will take a while until I can try this myself, but since 5 out of 6 snapshots are 100px wide (which is what I originally requested) and the 6th seems to accomodate my request from https://github.com/rstudio/shinytest2/issues/171#issuecomment-1128800258, this looks like the perfect solution to me. Looks like I can remove that workaround at some point :) thanks!

**Comment by bersbersbers** [comment]
(For the record, I haven't followed commits in a while, so I have no explanation why `snap9-prev-default.png` in https://github.com/rstudio/shinytest2/issues/171#issuecomment-1426263451 differs from the actual previous default `snap9.png` in my original post. But I am happy with it!)
